File: src/Entity/Vote.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;


/**
 * Vote
 *
 * @ORM\Table(name="vote", indexes={@ORM\Index(name="fk_vote_publication", columns={"idPublication"}), @ORM\Index(name="fk_vote_client", columns={"idClient"})})
 * @ORM\Entity(repositoryClass="App\Repository\VoteRepository")
 */
class Vote
{
    /**
     * @var \Client
     *
     * @ORM\Id()
     * @ORM\ManyToOne(targetEntity="Client")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idClient", referencedColumnName="idClient")
     * })
     */
    private $idclient;

    /**
     * @var \Publication
     *
     * @ORM\ManyToOne(targetEntity="Publication")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idPublication", referencedColumnName="idPublication")
     * })
     */
    private $idpublication;

    /**
     * @var string
     *
     * @ORM\Column(name="type", type="string", length=10, nullable=false)
     */
    private $type;

    public function getIdclient(): ?Client
    {
        return $this->idclient;
    }

    public function setIdclient(?Client $idclient): self
    {
        $this->idclient = $idclient;

        return $this;
    }

    public function getIdpublication(): ?Publication
    {
        return $this->idpublication;
    }

    public function setIdpublication(?Publication $pub): self
    {
        $this->idpublication = $pub;
        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }


}

File: GamerGeeksCommunity-Mobile/src/entities/Vote.java
package entities;

public class Vote {
    int id_client,id_publication;
    String type;

    public Vote(){
    }

    public Vote(int id_client, int id_publication, String type) {
        this.id_client = id_client;
        this.id_publication = id_publication;
        this.type = type;
    }

    public int getId_client() {
        return id_client;
    }

    public void setId_client(int id_client) {
        this.id_client = id_client;
    }

    public int getId_publication() {
        return id_publication;
    }

    public void setId_publication(int id_publication) {
        this.id_publication = id_publication;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "Vote{" +
                "id_client=" + id_client +
                ", id_publication=" + id_publication +
                ", type='" + type + '\'' +
                '}';
    }
}
